packages/sdk/src/restclient.ts

```diff
@@ -284,14 +284,14 @@ export class RestClient {
 
   // Makes a "smart query" on the contract, returns response verbatim (json.RawMessage)
   // Throws error if no such contract or invalid query format
-  public async queryContractSmart(address: string, query: object): Promise<unknown> {
+  public async queryContractSmart(address: string, query: object): Promise<string> {
```

Comment:
So. Would it be better if it returned a string like other calls, that you need to JSON.parse?

Cuz right now it parses it server side. Same as init_msg on the contract - the original object, not the string version

Reply:
There are multiple levels of parsing. The Rust `Result` level is parsed, but the binary data inside of `Ok(data)` is converted to a string. This diff reflect the fact that we always get a string here. At some point in the stack there must be a `Vec<u8>` to string conversion, that probably breaks when a contract decodes to send non-UTF8 binary data.

Reply:
Let's make a todo to fix/test that on wasmd side